deluge/ui/gtkui/mainwindow.py

```diff
@@ -117,15 +117,25 @@ def first_show(self):
             # Handle win32 WM_QUERYENDSESSION message so GUI can be shutdown cleanly.
             from win32gui import CallWindowProc, SetWindowLong
             from win32con import GWL_WNDPROC, WM_QUERYENDSESSION
+            from ctypes import windll, c_wchar_p
 
             def on_wndproc(hwnd, msg, wparam, lparam):
                 # log.debug("%s, %s, %s", msg, wParam, lParam)
+
+                ret = CallWindowProc(old_wndproc, hwnd, msg, wparam, lparam)
+
                 if msg == WM_QUERYENDSESSION:
-                    reactor.stop()
+                    self.window_handler = hwnd
```

Comment:
Would this be a more descriptive name: `self.win32_shutdown_block` 

Also it needs defined in MainWindow __init__ otherwise it will throw a `NameError` in gtkui shutdown.

Reply:
Also also should this only be set if ShutdownBlockReasonCreate returns non-zero?